src/normalgamma.jl

```diff
@@ -3,33 +3,38 @@
 # a reference.  Note that there were some typos in that document so the code
 # here may not correspond exactly.
 
-immutable NormalGamma <: Distribution
-    mu::Float64
-    nu::Float64     # scales precision of Normal
-    shape::Float64  
-    rate::Float64
+struct NormalGamma{T<:Real} <: ContinuousUnivariateDistribution
+    mu::T
+    nu::T     # scales precision of Normal
+    shape::T  
+    rate::T
 
-    function NormalGamma(mu::Real, nu::Real, sh::Real, r::Real)
+    function NormalGamma{T}(mu::T, nu::T, sh::T, r::T) where T<:Real
     	nu > zero(nu) && sh > zero(sh) && r > zero(r) || error("Both shape and scale must be positive")
-    	new(mu, nu, sh, r)
+    	new(T(mu), T(nu), T(sh), T(r))
     end
 end
 
+function NormalGamma(mu::Real, nu::Real, sh::Real, r::Real)
+    T = promote_type(typeof(mu), typeof(nu), typeof(sh), typeof(r))
+    return NormalGamma{T}(T(mu),T(nu),T(sh),T(r))
+end
+
 mu(d::NormalGamma) = d.mu
 nu(d::NormalGamma) = d.nu
 shape(d::NormalGamma) = d.shape
-scale(d::NormalGamma) = 1. / d.rate
+scale(d::NormalGamma{T}) where T = 1. / d.rate
```

Comment:
I'd say go ahead and drop support for versions prior to 0.6. The changes in 0.6 are significant enough that it doesn't make sense to try to keep supporting so many versions.